js/app/src/components/Audio/Audio.test.ts

```diff
@@ -0,0 +1,150 @@
+import { test, describe, assert, afterEach, vi } from "vitest";
+import { spy } from "tinyspy";
+import { cleanup, render } from "@gradio/tootils";
+import event from "@testing-library/user-event";
+import { setupi18n } from "../../i18n";
+
+import Audio from "./Audio.svelte";
+import type { LoadingStatus } from "../StatusTracker/types";
+
+const loading_status = {
+	eta: 0,
+	queue_position: 1,
+	queue_size: 1,
+	status: "complete" as LoadingStatus["status"],
+	scroll_to_output: false,
+	visible: true,
+	fn_index: 0,
+	show_progress: "full" as LoadingStatus["show_progress"]
+};
+
+describe("Audio", () => {
+	afterEach(() => cleanup());
+
+	test("renders provided value and label", async () => {
+		const { getByTestId, container, queryAllByText } = render(Audio, {
+			show_label: true,
+			loading_status,
+			mode: "dynamic",
+			value: {
+				name: "https://gradio-builds.s3.amazonaws.com/demo-files/audio_sample.wav",
+				data: null,
+				is_file: true
+			},
+			label: "Audio Component",
+			root: "foo",
+			root_url: null,
+			streaming: false,
+			pending: false,
+			name: "bar",
+			source: "upload"
+		});
+
+		assert.equal(
+			getByTestId("Audio Component-dynamic-audio").src,
+			"foo/file=https://gradio-builds.s3.amazonaws.com/demo-files/audio_sample.wav"
+		);
+		assert(queryAllByText("Audio Component").length, 1);
+	});
+
+	test("hides label", async () => {
+		const { queryAllByText } = render(Audio, {
+			show_label: false,
+			loading_status,
+			mode: "dynamic",
+			value: {
+				name: "https://gradio-builds.s3.amazonaws.com/demo-files/audio_sample.wav",
+				data: null,
+				is_file: true
+			},
+			label: "Audio Component",
+			root: "foo",
+			root_url: null,
+			streaming: false,
+			pending: false,
+			name: "bar",
+			source: "upload"
+		});
+
+		assert(queryAllByText("Audio Component").length, 0);
+	});
+
+	test("upload sets change event", async () => {
+		setupi18n();
```

Comment:
Took me forever to figure out I needed to initialize the i18n otherwise UploadText errors

Reply:
We should probably improve this.